After updating to 2.0 from 1.2.2 : git fetch exited with code 128

I'm not sure if the problem comes from OPAM or from the package owner, but it used to work well in 1.2.2.  After updating from 1.2.2 to 2.0, I received this error during  `opam update` part __Synchronising development packages__.

```
[ERROR] Could not synchronize /home/remy/.opam/4.05.0/.opam-switch/sources/js_of_ocaml-ppx_deriving_json from
        "git+ssh://git@github.com//ocsigen/js_of_ocaml.git":
        "/usr/bin/git fetch -q ssh://git@github.com//ocsigen/js_of_ocaml.git" exited with code 128
```

Then I ran the git command from .opam-switch/sources/js_of_ocaml-ppx_deriving_json

```
/usr/bin/git fetch -q ssh://git@github.com//ocsigen/js_of_ocaml.git
fatal: remote error: 
  /ocsigen/js_of_ocaml is not a valid repository name
  Email support@github.com for help
```

The problem seems to come from the repo name that has two slashes instead of one : 
`//ocsigen/js_of_ocaml.git`. The same situation appears with different pinned packages that have, or have not, their own git repo.

Here is one of the incriminated OPAM file. 
https://github.com/ocsigen/js_of_ocaml/blob/master/js_of_ocaml-ppx_deriving_json.opam


And also, just in case, here is my config :

```
# opam config report
# opam-version      2.0.0~beta5 
# self-upgrade      no
# system            arch=x86_64 os=linux os-distribution=fedora os-version=27
# solver            builtin-mccs+glpk
# install-criteria  -removed,-count[version-lag,request],-count[version-lag,changed],-changed
# upgrade-criteria  -removed,-count[version-lag,solution],-new
# jobs              4
# repositories      1 (http) (default repo at 2018-02-06 10:04)
# pinned            26 (git)
# current-switch    4.05.0
```

--- a/src/core/opamUrl.ml
+++ b/src/core/opamUrl.ml
@@ -88,6 +88,7 @@ let looks_like_ssh_path =
     Re.(compile @@ seq [
         group @@ repn (diff any (set "/:")) 2 None;
         char ':';
+        opt @@ char '/';
         opt @@ group @@ seq [
           alt [
             diff any digit;

